kdbplus/src/ipc/connection.rs

```diff
@@ -1275,30 +1261,29 @@ async fn decompress(compressed: Vec<u8>, encoding: u8) -> Vec<u8> {
             ) - 8
         }
     };
-    let mut decompressed: Vec<u8> = Vec::with_capacity(size as usize);
     // Assure that vector is filled with 0
-    decompressed.resize(size as usize, 0_u8);
+    let mut decompressed: Vec<u8> = vec![0_u8; size as usize];
 
     // Start index of compressed body.
     // 8 bytes have already been removed as header
     let mut d = 4;
     let mut aa = [0_i32; 256];
     while s < decompressed.len() {
         if i == 0 {
-            f = (0xff & compressed[d]) as usize;
```

Comment:
The compression algorithm is referring to [Kx Java adaptor](https://github.com/KxSystems/javakdb/blob/4a6379e08c0fab14b2ce505d47ab026bdd24ad60/javakdb/src/main/java/com/kx/c.java#L502). There `0xff` is used as mask.

Reply:
![image](https://github.com/diamondrod/kdbplus/assets/68485672/a4b49e75-e395-4758-bec6-983ac8eda58e)

Reply:
If Clippy is wrong, I can revert the change and suppress the lint.

Reply:
0xff is equivalent to 0b1111_1111, running an AND operation on that and another variable will just give you that other variables


0b1 & 0b1 == 0b1
0b1 & 0b0 == 0b0